akamai/cache-client.go

```diff
@@ -0,0 +1,255 @@
+package akamai
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/golang.org/x/crypto/ocsp"
+
+	blog "github.com/letsencrypt/boulder/log"
+)
+
+const (
+	purgePath       = "/ccu/v2/queues/default"
+	timestampFormat = "20060102T15:04:05-0700"
+)
+
+type purgeRequest struct {
+	Objects []string `json:"objects"`
+}
+
+type purgeResponse struct {
+	HTTPStatus       int    `json:"httpStatus"`
+	Detail           string `json:"detail"`
+	EstimatedSeconds int    `json:"estimatedSeconds"`
+	PurgeID          string `json:"purgeId"`
+}
+
+// CachePurgeClient talks to the Akamai CCU REST API. It is safe to make concurrent
+// purge requests.
+type CachePurgeClient struct {
+	client       *http.Client
+	apiEndpoint  string
+	apiHost      string
+	apiScheme    string
+	clientToken  string
+	clientSecret string
+	accessToken  string
+	issuer       *x509.Certificate
+	retries      int
+	retryBackoff time.Duration
+	log          *blog.AuditLogger
+	stats        statsd.Statter
+	clk          clock.Clock
+}
+
+type errFatal string
```

Comment:
Please document what errFatal means in this context. My first guess was that it means an error which should terminate the program, but I'm pretty sure that's not correct.